pkg/etcdcli/health.go

```diff
@@ -80,30 +80,17 @@ func GetMemberHealth(ctx context.Context, etcdMembers []*etcdserverpb.Member) me
 	return memberHealth
 }
 
-func checkSingleMemberHealth(ctx context.Context, member *etcdserverpb.Member) healthCheck {
-	// If the endpoint is for a learner member then we should skip testing the connection
-	// via the member list call as learners don't support that.
-	// The learner's connection would get tested in the health check below
-	skipConnectionTest := false
-	if member.IsLearner {
-		skipConnectionTest = true
-	}
-	cli, err := newEtcdClientWithClientOpts([]string{member.ClientURLs[0]}, skipConnectionTest)
-	if err != nil {
-		return healthCheck{
-			Member:  member,
-			Healthy: false,
-			Error:   fmt.Errorf("create client failure: %w", err)}
-	}
+func checkSingleMemberHealth(ctx context.Context, cli *clientv3.Client, member *etcdserverpb.Member) healthCheck {
+	originalEndpoints := cli.Endpoints()
```

Comment:
nit: You could do:
```go
defer cli.SetEndpoints(cli.Endpoints())
```
and cli.Endpoints() will be cached and that cached version will be used at deferred execution

Reply:
I leave this for now to not have your lgtm being wasted - thanks for the research, I was curious about this too :)

Reply:
No worries! Here's the proof! https://go.dev/play/p/Cgzi2FAO45G

The great package github.com/pkg/profile abuses this to make profiling easier to set up.